src/platform/meteorlake/include/platform/lib/ams.h

```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ */
+
+#ifdef __SOF_LIB_AMS_H__
+
+#ifndef __PLATFORM_LIB_AMS_H__
+#define __PLATFORM_LIB_AMS_H__
+
+/* max number of message UUIDs */
+#define AMS_SERVICE_UUID_TABLE_SIZE 16
+/* max number of async message routes */
+#define AMS_ROUTING_TABLE_SIZE 16
```

Comment:
is this related to the other  AMS_SERVICE_UUID_TABLE_SIZE above?

I am also not sure I understand the concept of 'route' given that the communication seems to be point-to-point with the core and instance ID specified.

Reply:
The more message types the more routing entries, but there's no 1:1 mapping in the values. 
Async messages support 1:1, 1:N, M:1, M:N communication, this does not have to be point to point. The "routing" in name specifies the function routing table performs. That's the list of registered consumer entries that is walked through while routing a message. 
So for a given message `entries msg type is checked` -> `if match, check module id and instance id` -> `if match, check core id` -> `if same core, invoke callback / if different core, sent through idc.` Single consumer (module) can be listed there multiple times for multiple message ids it should recveive.
So yes, it could be called sth like "consumer_entries_table", but the main point of having it is not to hold consumer entries, but to route messages between them. I hardly believe that's a better naming.

Reply:
I really don't see how you can have a M:N mapping, since there is ONE producer_register. it's 1:N or you are missing a lot of explanations in this PR.